sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -337,6 +337,7 @@ class SQLQueryTestSuite extends QueryTest with SharedSparkSession {
         localSparkSession.conf.set(SQLConf.ANSI_ENABLED.key, true)
       case _ =>
     }
+    localSparkSession.conf.set(SQLConf.DATETIME_JAVA8API_ENABLED.key, true)
```

Comment:
I think this is a bug in `HiveResult`, not just this test suite.

When we produce string result, we must stick with the Proleptic Gregorian calendar and call `LocalDate/Instant.toString`. I think we should update `HiveResult.hiveResultString` and enable java 8 datetime API before calling `collect`.

Reply:
HiveResult is used in `hive-thriftserver` too. If I set `spark.sql.datetime.java8API.enabled` to `true` in `HiveResult.hiveResultString`, this might effect on the Thrift server. Though I can correct expected results in the tests, but Thrift server may require broader changes.

Reply:
Can we open a new PR to do it? I believe we need to do this after switching to Proleptic Gregorian calendar. We should avoid using different calendars inconsistently.

Reply:
Sure, I will do that after this PR will be merged because I will need the changes, most likely.